File: server/metrics/generate_docs_test.py
#!/usr/bin/env python3

import unittest
from server.metrics.generate_docs import DocsGenerator


class TestDocsGenerator(unittest.TestCase):
    def test_parse_label_constant_with_single_space(self):
        """Test parsing label constant with single space around ="""
        gen = DocsGenerator("/dev/null")
        gen.state = "LABEL_CONSTANTS"
        gen.process_line(0, '    StatusLabel = "status"')
        self.assertIn("StatusLabel", gen.label_constants)
        self.assertEqual(gen.label_constants["StatusLabel"]["value"], "status")

    def test_parse_label_constant_with_multiple_spaces(self):
        """Test parsing label constant with gofmt-aligned multiple spaces"""
        gen = DocsGenerator("/dev/null")
        gen.state = "LABEL_CONSTANTS"
        gen.process_line(0, '    StatusHumanReadableLabel         = "status"')
        self.assertIn("StatusHumanReadableLabel", gen.label_constants)
        self.assertEqual(gen.label_constants["StatusHumanReadableLabel"]["value"], "status")

if __name__ == "__main__":
    unittest.main()

File: server/metrics/generate_docs/generate_docs_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestParseLabelConstantWithSingleSpace(t *testing.T) {
	gen := newDocsGenerator("/dev/null")
	gen.state = "LABEL_CONSTANTS"
	err := gen.processLine(0, `    StatusLabel = "status"`)
	require.NoError(t, err)
	constInfo, ok := gen.labelConstants["StatusLabel"]
	require.True(t, ok, "expected StatusLabel to be in labelConstants")
	assert.Equal(t, "status", constInfo.value)
}

func TestParseLabelConstantWithMultipleSpaces(t *testing.T) {
	gen := newDocsGenerator("/dev/null")
	gen.state = "LABEL_CONSTANTS"
	err := gen.processLine(0, `    StatusHumanReadableLabel         = "status"`)
	require.NoError(t, err)
	constInfo, ok := gen.labelConstants["StatusHumanReadableLabel"]
	require.True(t, ok, "expected StatusHumanReadableLabel to be in labelConstants")
	assert.Equal(t, "status", constInfo.value)
}
